src/mlpack/methods/kde/kde_rules.hpp

```diff
@@ -25,6 +25,10 @@ namespace kde {
 template<typename MetricType, typename KernelType, typename TreeType>
 class KDERules
 {
+ private:
+  // SFINAE Check if kernel has bandwidth function.
+  HAS_MEM_FUNC(Bandwidth, HasBandwidth);
```

Comment:
I think maybe this is extra; I don't see it used. (Correct me if I missed it. :))